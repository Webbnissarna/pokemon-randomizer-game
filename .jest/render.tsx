import { render as rtlRender } from "@testing-library/react";
import { InitializeColorMode, ThemeProvider } from "theme-ui";

import theme from "../utils/theme";

import "./style.css";

export function render(ui: React.ReactElement) {
  return rtlRender(
    <>
      <InitializeColorMode />
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </>
  );
}
