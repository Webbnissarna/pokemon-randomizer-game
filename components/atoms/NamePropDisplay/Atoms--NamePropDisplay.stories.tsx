import NamePropDisplay from ".";

interface ControlProps {
  name: string;
  mainProp: string;
  optionalProp: string;
}

export function Default({ name, mainProp, optionalProp }: ControlProps) {
  return <NamePropDisplay name={name} prop={{ mainProp, optionalProp }} />;
}

Default.args = {
  name: "Pikachu",
  mainProp: "ピカチュウ",
  optionalProp: "Pikachū",
} as ControlProps;

Default.argTypes = {
  size: {
    options: ["sm", "md", "lg"],
    control: { type: "select" },
  },
  variant: {
    options: ["positive", "negative", "neutral", "secondary", "tertiary"],
    control: { type: "radio" },
    defaultValue: "positive",
  },
};
