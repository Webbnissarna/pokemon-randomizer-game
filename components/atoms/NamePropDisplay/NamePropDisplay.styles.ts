import type { ThemeUIStyleObject } from "theme-ui";

export const containerStyle: ThemeUIStyleObject = {
  flexDirection: "column",
  textAlign: "center",
};
