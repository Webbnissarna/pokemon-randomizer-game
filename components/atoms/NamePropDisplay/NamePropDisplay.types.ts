import type { GameProp } from "../../../utils/GamePropSystem/GamePropSystem.types";

export interface NamePropDisplayProps {
  name?: string;
  prop: GameProp;
}
