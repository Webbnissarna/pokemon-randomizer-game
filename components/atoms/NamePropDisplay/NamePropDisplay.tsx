import { Flex, Heading, Text } from "theme-ui";
import { containerStyle } from "./NamePropDisplay.styles";

import type { NamePropDisplayProps } from "./NamePropDisplay.types";

export default function NamePropDisplay({
  name,
  prop,
}: NamePropDisplayProps): JSX.Element {
  return (
    <Flex sx={containerStyle}>
      <Heading as="h1">{name}</Heading>
      <Heading as="h2">{prop.mainProp}</Heading>
      {prop.optionalProp && <Text>{prop.optionalProp}</Text>}
    </Flex>
  );
}
