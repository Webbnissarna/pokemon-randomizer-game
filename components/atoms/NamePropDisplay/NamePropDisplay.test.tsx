import { render } from "../../../.jest/render";
import NamePropDisplay from ".";

describe("NamePropDisplay", () => {
  it("renders", () => {
    const { asFragment } = render(<NamePropDisplay prop={{ mainProp: "A" }} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
