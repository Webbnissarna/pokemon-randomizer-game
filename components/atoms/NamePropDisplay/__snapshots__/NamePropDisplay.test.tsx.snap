// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NamePropDisplay renders 1`] = `
<DocumentFragment>
  <script>
    (function() { try {
  var mode = localStorage.getItem('theme-ui-color-mode');
  if (!mode) return
  document.documentElement.classList.add('theme-ui-' + mode);
} catch (e) {} })();
  </script>
  <div
    class="css-1eqdum8-Box"
  >
    <h1
      class="css-njhccr-Box"
    />
    <h2
      class="css-njhccr-Box"
    >
      A
    </h2>
  </div>
</DocumentFragment>
`;
